src/main/java/org/opentripplanner/model/plan/FareLeg.java

```diff
@@ -0,0 +1,19 @@
+package org.opentripplanner.model.plan;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.opentripplanner.transit.model.site.FareZone;
+
+public interface FareLeg extends Leg {
```

Comment:
I'm not so fond of adding a new interface here, especially since a street leg could also have fares, eg. for rental vehicles.

Reply:
Where do you recommend that this code lives?

BTW, this doesn't have any effect on vehicle rental calculation. There it works like this: https://github.com/opentripplanner/OpenTripPlanner/blob/17e16dc22573b3807a4d010261f9b5001597e455/src/main/java/org/opentripplanner/routing/fares/impl/TimeBasedVehicleRentalFareService.java#L36-L46

But yes, that interface name is confusing.

Reply:
I think we could just expand the `Leg` interface with `Set<FareZone> getFareZones()`, and either just copy this code in two places, or create a new utility class to host it.

Reply:
done